spring-rabbit/src/main/java/org/springframework/amqp/rabbit/aot/RabbitRuntimeHintsRegistrar.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit.aot;
+
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.amqp.rabbit.connection.ChannelProxy;
+import org.springframework.aop.SpringProxy;
+import org.springframework.aop.framework.Advised;
+import org.springframework.aot.hint.ProxyHints;
+import org.springframework.aot.hint.RuntimeHints;
+import org.springframework.aot.hint.RuntimeHintsRegistrar;
+import org.springframework.aot.hint.TypeReference;
+import org.springframework.core.DecoratingProxy;
+import org.springframework.core.annotation.SynthesizedAnnotation;
+import org.springframework.lang.Nullable;
+
+/**
+ * {@link RuntimeHintsRegistrar} for spring-rabbit.
+ *
+ * @author Gary Russell
+ * @since 3.0
+ *
+ */
+public class RabbitRuntimeHintsRegistrar implements RuntimeHintsRegistrar {
+
+	@Override
+	public void registerHints(RuntimeHints hints, @Nullable	ClassLoader classLoader) {
+		ProxyHints proxyHints = hints.proxies();
+		proxyHints.registerJdkProxy(ChannelProxy.class);
```

Comment:
Gary, don't we need also these interfaces for each proxy: `SpringProxy.class, Advised.class, DecoratingProxy.class` ?
See my PR in SI: https://github.com/spring-projects/spring-integration/pull/3832/files#diff-b471878878aa8838786d4a93b19f8014d497db6ace582572f834da814720e01bR150.

Also see this new API in SF `6.0`: 
```
runtimeHints.proxies().registerJdkProxy(AopProxyUtils.completeJdkProxyInterfaces(proxyInterfaces))
```

Reply:
I just replicated what we have in spring-native (which works).

https://github.com/spring-projects-experimental/spring-native/blob/2d1fc19c3833e6ce2bd4d342504b180a88407572/spring-native-configuration/src/main/java/org/springframework/amqp/rabbit/annotation/RabbitHints.java#L28-L35

I will find out soon because the compile problem has now been resolved.

Reply:
I see. Our logic for this proxy is not about Spring. Agreed:
```
		if (ConfirmType.CORRELATED.equals(this.confirmType) || this.publisherReturns) {
			interfaces = new Class<?>[] { ChannelProxy.class, PublisherCallbackChannel.class };
		}
		else {
			interfaces = new Class<?>[] { ChannelProxy.class };
		}
		return (ChannelProxy) Proxy.newProxyInstance(ChannelProxy.class.getClassLoader(),
				interfaces, new CachedChannelInvocationHandler(connection, targetChannel, channelList,
						transactional));
```
However I I guess we need to register one more proxy for the `ChannelProxy.class` and `PublisherCallbackChannel.class` combination according the logic of that `CachingConnectionFactory` class.
The one for `RabbitListener` annotation is OK. However it might be something like:
```
			RuntimeHintsUtils.registerAnnotation(hints, Scheduled.class);
```
instead ?